### ISSUE TITLE
Insetting APIs for Rect

### ISSUE THREAD

**Issue by andymatuschak** [Khan/Prototope#50]
Would like to be able to do things like:

```
var someRect = new Rect({x: 5, y: 10, width: 20, height: 30})
var x = someRect.inset({top: 30, right: 15})
var y = someRect.inset({vertical: 30, horizontal: 15})
var z = someRect.inset(20)
```

**Comment by jbrennan** [comment]
Wait, are the return values supposed to be rects or are they x/y/z coords?

**Comment by jbrennan** [comment]
Wait, are the return values supposed to be rects or are they x/y/z coords?

**Comment by jbrennan** [comment]
Wait, are the return values supposed to be rects or are they x/y/z coords?

**Comment by NachoSoto** [comment]
I keep reading this as `Inserting APIs for React`.

**Comment by spicyj** [comment]
Me too.

**Comment by saniul** [comment]
> Wait, are the return values supposed to be rects or are they x/y/z coords?

I understood those to be rects

**Comment by saniul** [comment]
Feedback wanted for https://github.com/Khan/Prototope/pull/58

**Comment by jbrennan** [comment]
Fixed by #58
